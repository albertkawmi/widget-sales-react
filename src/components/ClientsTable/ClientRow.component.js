import React from 'react'
import ClickableRow from '../ClickableRow'
import LinkButton from '../LinkButton'

// Can't use an arrow function as Enzyme renders it 'Unknown'
function ClientRow({
  id,
  firstName,
  lastName,
  company,
  telephone,
  country,
  isSelected
}) {
  return (
    <ClickableRow
      key={isSelected}
      isClickable={!isSelected}
      to={`/clients/${id}`}
      className={isSelected ? 'selected' : ''}
    >
      <td>
        <LinkButton
          to={`/clients/${id}`}
          text="Select"
          options={{ disabled: isSelected }}
          buttonClassName="select-client-btn"
        />
      </td>
      <td>{id}</td>
      <td>{firstName}</td>
      <td>{lastName}</td>
      <td>{company}</td>
      <td>{telephone}</td>
      <td>{country}</td>
    </ClickableRow>
  )
}

export default ClientRow
