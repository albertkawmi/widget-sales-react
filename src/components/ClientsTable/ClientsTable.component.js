import React from 'react'
import { Link } from 'react-router-dom'

const ClientsTable = ({ clients, selectedId }) => (
  <table cellSpacing="0">
    <thead>
      <tr>
        <th>ID</th>
        <th>First Name</th>
        <th>Last Name</th>
        <th>Company</th>
        <th>Telephone</th>
        <th>Country</th>
        <th>&nbsp;</th>
      </tr>
    </thead>
    <tbody>
      {clients.map(client => <ClientRow
        key={client.id}
        isSelected={selectedId === client.id}
        {...client}
      />)}
    </tbody>
  </table>
)

// Can't use an arrow function as Enzyme renders it 'Unknown'
export function ClientRow({
  id,
  firstName,
  lastName,
  company,
  telephone,
  country,
  isSelected
}) {
  return (
    <tr className={isSelected ? 'selected' : ''}>
      <td>{id}</td>
      <td>{firstName}</td>
      <td>{lastName}</td>
      <td>{company}</td>
      <td>{telephone}</td>
      <td>{country}</td>
      <td>
        <Link to={`/clients/${id}`}>
          <button disabled={isSelected}>Select</button>
        </Link>
      </td>
    </tr>
  )
}

export default ClientsTable
