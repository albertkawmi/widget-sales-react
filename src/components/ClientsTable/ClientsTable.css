.sales {
  padding: 1rem;
  margin-top: 2rem;
  background: whitesmoke;
  border: solid 1px silver;
}

.sale-cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0;
  margin: 0;
}

.sale-card {
  box-sizing: border-box;
  list-style-type: none;
  margin: 1rem;
  background: white;
  border: solid 1px whitesmoke;
  box-shadow: 1px 1px silver;
  padding: 1rem;
  flex-basis: 29%;
}
