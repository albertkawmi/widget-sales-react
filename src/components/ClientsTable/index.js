import ClientsTable from './ClientsTable.component'
export default ClientsTable
