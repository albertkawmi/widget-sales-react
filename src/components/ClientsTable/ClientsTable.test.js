/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ClientsTable from './ClientsTable.component'
import ClientRow from './ClientRow.component'
import clientsData from '../../../mocks/clients.json'

Enzyme.configure({ adapter: new Adapter() })

describe('<ClientsTable />', () => {
  it('renders a spinner if it has no data', () => {
    const table = Enzyme.shallow(<ClientsTable clients={[]} />)
    expect(
      table.find('Spinner').exists()
    ).toBe(true)
  })

  it('renders clients data in a table', () => {
    const table = Enzyme.shallow(<ClientsTable clients={clientsData} />)
    expect(table).toMatchSnapshot()
  })

  it('highlights the selected row', () => {
    const table = Enzyme.shallow(
      <ClientsTable clients={clientsData} selectedId={3} />
    )

    const selectedRows = table.findWhere(
      row => row.prop('isSelected') === true
    )

    expect(selectedRows.length).toBe(1)
  })

  it('it does not highlight any row if there is no matching id', () => {
    const table = Enzyme.shallow(
      // clients.json contains 10 rows, so selectedId 11 will not match
      <ClientsTable clients={clientsData} selectedId={11} />
    )
    expect(
      table.findWhere(row => row.prop('isSelected') === true).length
    ).toBe(0)
  })

  it('has a link to select each row', () => {
    const row = Enzyme.shallow(
      <ClientRow isSelected={false} {...clientsData[0]} />
    )
    expect(
      row.find('LinkButton').prop('to')
    ).toBe('/clients/1')
  })

  it('renders even if clients prop is undefined', () => {
    expect(
      Enzyme.shallow(<ClientsTable />)
    ).toBeTruthy()
  })
})
