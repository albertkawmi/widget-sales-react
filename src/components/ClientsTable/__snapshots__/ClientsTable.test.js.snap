// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientsTable /> renders clients data in a table 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClientsTable
    clients={
        Array [
            Object {
              "company": "Dynazzy",
              "country": "Nigeria",
              "firstName": "Mariejeanne",
              "id": 1,
              "lastName": "Lush",
              "telephone": "+234 494 203 0408",
            },
            Object {
              "company": "Bluejam",
              "country": "Zimbabwe",
              "firstName": "Jonas",
              "id": 2,
              "lastName": "Jeannaud",
              "telephone": "+263 706 597 3567",
            },
            Object {
              "company": "Divape",
              "country": "Philippines",
              "firstName": "Raynell",
              "id": 3,
              "lastName": "Manning",
              "telephone": "+63 672 620 4807",
            },
            Object {
              "company": "Brightbean",
              "country": "Sweden",
              "firstName": "Orlan",
              "id": 4,
              "lastName": "Westbrook",
              "telephone": "+46 709 539 1100",
            },
            Object {
              "company": "Wikizz",
              "country": "Indonesia",
              "firstName": "Ivory",
              "id": 5,
              "lastName": "Plaister",
              "telephone": "+62 443 276 2504",
            },
            Object {
              "company": "Avamm",
              "country": "China",
              "firstName": "Lindsey",
              "id": 6,
              "lastName": "Revie",
              "telephone": "+86 341 167 4508",
            },
            Object {
              "company": "Mita",
              "country": "Indonesia",
              "firstName": "Constantin",
              "id": 7,
              "lastName": "Tyer",
              "telephone": "+62 201 966 3247",
            },
            Object {
              "company": "Quimm",
              "country": "Poland",
              "firstName": "Letizia",
              "id": 8,
              "lastName": "Emeney",
              "telephone": "+48 944 486 3972",
            },
            Object {
              "company": "Topicshots",
              "country": "Portugal",
              "firstName": "Rivkah",
              "id": 9,
              "lastName": "Sabberton",
              "telephone": "+351 675 609 7738",
            },
            Object {
              "company": "Browsedrive",
              "country": "Dominican Republic",
              "firstName": "Jaine",
              "id": 10,
              "lastName": "Hubback",
              "telephone": "+1 411 500 2183",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "cellSpacing": "0",
      "children": Array [
        <thead>
          <tr>
                    <th>
                              ID
                    </th>
                    <th>
                              First Name
                    </th>
                    <th>
                              Last Name
                    </th>
                    <th>
                              Company
                    </th>
                    <th>
                              Telephone
                    </th>
                    <th>
                              Country
                    </th>
                    <th>
                               
                    </th>
          </tr>
</thead>,
        <tbody>
          <ClientRow
                    company="Dynazzy"
                    country="Nigeria"
                    firstName="Mariejeanne"
                    id={1}
                    isSelected={false}
                    lastName="Lush"
                    telephone="+234 494 203 0408"
          />
          <ClientRow
                    company="Bluejam"
                    country="Zimbabwe"
                    firstName="Jonas"
                    id={2}
                    isSelected={false}
                    lastName="Jeannaud"
                    telephone="+263 706 597 3567"
          />
          <ClientRow
                    company="Divape"
                    country="Philippines"
                    firstName="Raynell"
                    id={3}
                    isSelected={false}
                    lastName="Manning"
                    telephone="+63 672 620 4807"
          />
          <ClientRow
                    company="Brightbean"
                    country="Sweden"
                    firstName="Orlan"
                    id={4}
                    isSelected={false}
                    lastName="Westbrook"
                    telephone="+46 709 539 1100"
          />
          <ClientRow
                    company="Wikizz"
                    country="Indonesia"
                    firstName="Ivory"
                    id={5}
                    isSelected={false}
                    lastName="Plaister"
                    telephone="+62 443 276 2504"
          />
          <ClientRow
                    company="Avamm"
                    country="China"
                    firstName="Lindsey"
                    id={6}
                    isSelected={false}
                    lastName="Revie"
                    telephone="+86 341 167 4508"
          />
          <ClientRow
                    company="Mita"
                    country="Indonesia"
                    firstName="Constantin"
                    id={7}
                    isSelected={false}
                    lastName="Tyer"
                    telephone="+62 201 966 3247"
          />
          <ClientRow
                    company="Quimm"
                    country="Poland"
                    firstName="Letizia"
                    id={8}
                    isSelected={false}
                    lastName="Emeney"
                    telephone="+48 944 486 3972"
          />
          <ClientRow
                    company="Topicshots"
                    country="Portugal"
                    firstName="Rivkah"
                    id={9}
                    isSelected={false}
                    lastName="Sabberton"
                    telephone="+351 675 609 7738"
          />
          <ClientRow
                    company="Browsedrive"
                    country="Dominican Republic"
                    firstName="Jaine"
                    id={10}
                    isSelected={false}
                    lastName="Hubback"
                    telephone="+1 411 500 2183"
          />
</tbody>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
                        ID
            </th>
            <th>
                        First Name
            </th>
            <th>
                        Last Name
            </th>
            <th>
                        Company
            </th>
            <th>
                        Telephone
            </th>
            <th>
                        Country
            </th>
            <th>
                         
            </th>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                ID
</th>,
              <th>
                First Name
</th>,
              <th>
                Last Name
</th>,
              <th>
                Company
</th>,
              <th>
                Telephone
</th>,
              <th>
                Country
</th>,
              <th>
                 
</th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ID",
              },
              "ref": null,
              "rendered": "ID",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First Name",
              },
              "ref": null,
              "rendered": "First Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last Name",
              },
              "ref": null,
              "rendered": "Last Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Company",
              },
              "ref": null,
              "rendered": "Company",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Telephone",
              },
              "ref": null,
              "rendered": "Telephone",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Country",
              },
              "ref": null,
              "rendered": "Country",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " ",
              },
              "ref": null,
              "rendered": " ",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ClientRow
              company="Dynazzy"
              country="Nigeria"
              firstName="Mariejeanne"
              id={1}
              isSelected={false}
              lastName="Lush"
              telephone="+234 494 203 0408"
/>,
            <ClientRow
              company="Bluejam"
              country="Zimbabwe"
              firstName="Jonas"
              id={2}
              isSelected={false}
              lastName="Jeannaud"
              telephone="+263 706 597 3567"
/>,
            <ClientRow
              company="Divape"
              country="Philippines"
              firstName="Raynell"
              id={3}
              isSelected={false}
              lastName="Manning"
              telephone="+63 672 620 4807"
/>,
            <ClientRow
              company="Brightbean"
              country="Sweden"
              firstName="Orlan"
              id={4}
              isSelected={false}
              lastName="Westbrook"
              telephone="+46 709 539 1100"
/>,
            <ClientRow
              company="Wikizz"
              country="Indonesia"
              firstName="Ivory"
              id={5}
              isSelected={false}
              lastName="Plaister"
              telephone="+62 443 276 2504"
/>,
            <ClientRow
              company="Avamm"
              country="China"
              firstName="Lindsey"
              id={6}
              isSelected={false}
              lastName="Revie"
              telephone="+86 341 167 4508"
/>,
            <ClientRow
              company="Mita"
              country="Indonesia"
              firstName="Constantin"
              id={7}
              isSelected={false}
              lastName="Tyer"
              telephone="+62 201 966 3247"
/>,
            <ClientRow
              company="Quimm"
              country="Poland"
              firstName="Letizia"
              id={8}
              isSelected={false}
              lastName="Emeney"
              telephone="+48 944 486 3972"
/>,
            <ClientRow
              company="Topicshots"
              country="Portugal"
              firstName="Rivkah"
              id={9}
              isSelected={false}
              lastName="Sabberton"
              telephone="+351 675 609 7738"
/>,
            <ClientRow
              company="Browsedrive"
              country="Dominican Republic"
              firstName="Jaine"
              id={10}
              isSelected={false}
              lastName="Hubback"
              telephone="+1 411 500 2183"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "company": "Dynazzy",
              "country": "Nigeria",
              "firstName": "Mariejeanne",
              "id": 1,
              "isSelected": false,
              "lastName": "Lush",
              "telephone": "+234 494 203 0408",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "company": "Bluejam",
              "country": "Zimbabwe",
              "firstName": "Jonas",
              "id": 2,
              "isSelected": false,
              "lastName": "Jeannaud",
              "telephone": "+263 706 597 3567",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "company": "Divape",
              "country": "Philippines",
              "firstName": "Raynell",
              "id": 3,
              "isSelected": false,
              "lastName": "Manning",
              "telephone": "+63 672 620 4807",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "company": "Brightbean",
              "country": "Sweden",
              "firstName": "Orlan",
              "id": 4,
              "isSelected": false,
              "lastName": "Westbrook",
              "telephone": "+46 709 539 1100",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "company": "Wikizz",
              "country": "Indonesia",
              "firstName": "Ivory",
              "id": 5,
              "isSelected": false,
              "lastName": "Plaister",
              "telephone": "+62 443 276 2504",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "function",
            "props": Object {
              "company": "Avamm",
              "country": "China",
              "firstName": "Lindsey",
              "id": 6,
              "isSelected": false,
              "lastName": "Revie",
              "telephone": "+86 341 167 4508",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "function",
            "props": Object {
              "company": "Mita",
              "country": "Indonesia",
              "firstName": "Constantin",
              "id": 7,
              "isSelected": false,
              "lastName": "Tyer",
              "telephone": "+62 201 966 3247",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "function",
            "props": Object {
              "company": "Quimm",
              "country": "Poland",
              "firstName": "Letizia",
              "id": 8,
              "isSelected": false,
              "lastName": "Emeney",
              "telephone": "+48 944 486 3972",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "function",
            "props": Object {
              "company": "Topicshots",
              "country": "Portugal",
              "firstName": "Rivkah",
              "id": 9,
              "isSelected": false,
              "lastName": "Sabberton",
              "telephone": "+351 675 609 7738",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "function",
            "props": Object {
              "company": "Browsedrive",
              "country": "Dominican Republic",
              "firstName": "Jaine",
              "id": 10,
              "isSelected": false,
              "lastName": "Hubback",
              "telephone": "+1 411 500 2183",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "cellSpacing": "0",
        "children": Array [
          <thead>
            <tr>
                        <th>
                                    ID
                        </th>
                        <th>
                                    First Name
                        </th>
                        <th>
                                    Last Name
                        </th>
                        <th>
                                    Company
                        </th>
                        <th>
                                    Telephone
                        </th>
                        <th>
                                    Country
                        </th>
                        <th>
                                     
                        </th>
            </tr>
</thead>,
          <tbody>
            <ClientRow
                        company="Dynazzy"
                        country="Nigeria"
                        firstName="Mariejeanne"
                        id={1}
                        isSelected={false}
                        lastName="Lush"
                        telephone="+234 494 203 0408"
            />
            <ClientRow
                        company="Bluejam"
                        country="Zimbabwe"
                        firstName="Jonas"
                        id={2}
                        isSelected={false}
                        lastName="Jeannaud"
                        telephone="+263 706 597 3567"
            />
            <ClientRow
                        company="Divape"
                        country="Philippines"
                        firstName="Raynell"
                        id={3}
                        isSelected={false}
                        lastName="Manning"
                        telephone="+63 672 620 4807"
            />
            <ClientRow
                        company="Brightbean"
                        country="Sweden"
                        firstName="Orlan"
                        id={4}
                        isSelected={false}
                        lastName="Westbrook"
                        telephone="+46 709 539 1100"
            />
            <ClientRow
                        company="Wikizz"
                        country="Indonesia"
                        firstName="Ivory"
                        id={5}
                        isSelected={false}
                        lastName="Plaister"
                        telephone="+62 443 276 2504"
            />
            <ClientRow
                        company="Avamm"
                        country="China"
                        firstName="Lindsey"
                        id={6}
                        isSelected={false}
                        lastName="Revie"
                        telephone="+86 341 167 4508"
            />
            <ClientRow
                        company="Mita"
                        country="Indonesia"
                        firstName="Constantin"
                        id={7}
                        isSelected={false}
                        lastName="Tyer"
                        telephone="+62 201 966 3247"
            />
            <ClientRow
                        company="Quimm"
                        country="Poland"
                        firstName="Letizia"
                        id={8}
                        isSelected={false}
                        lastName="Emeney"
                        telephone="+48 944 486 3972"
            />
            <ClientRow
                        company="Topicshots"
                        country="Portugal"
                        firstName="Rivkah"
                        id={9}
                        isSelected={false}
                        lastName="Sabberton"
                        telephone="+351 675 609 7738"
            />
            <ClientRow
                        company="Browsedrive"
                        country="Dominican Republic"
                        firstName="Jaine"
                        id={10}
                        isSelected={false}
                        lastName="Hubback"
                        telephone="+1 411 500 2183"
            />
</tbody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                            ID
              </th>
              <th>
                            First Name
              </th>
              <th>
                            Last Name
              </th>
              <th>
                            Company
              </th>
              <th>
                            Telephone
              </th>
              <th>
                            Country
              </th>
              <th>
                             
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  ID
</th>,
                <th>
                  First Name
</th>,
                <th>
                  Last Name
</th>,
                <th>
                  Company
</th>,
                <th>
                  Telephone
</th>,
                <th>
                  Country
</th>,
                <th>
                   
</th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ID",
                },
                "ref": null,
                "rendered": "ID",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "First Name",
                },
                "ref": null,
                "rendered": "First Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Last Name",
                },
                "ref": null,
                "rendered": "Last Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Company",
                },
                "ref": null,
                "rendered": "Company",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Telephone",
                },
                "ref": null,
                "rendered": "Telephone",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Country",
                },
                "ref": null,
                "rendered": "Country",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " ",
                },
                "ref": null,
                "rendered": " ",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ClientRow
                company="Dynazzy"
                country="Nigeria"
                firstName="Mariejeanne"
                id={1}
                isSelected={false}
                lastName="Lush"
                telephone="+234 494 203 0408"
/>,
              <ClientRow
                company="Bluejam"
                country="Zimbabwe"
                firstName="Jonas"
                id={2}
                isSelected={false}
                lastName="Jeannaud"
                telephone="+263 706 597 3567"
/>,
              <ClientRow
                company="Divape"
                country="Philippines"
                firstName="Raynell"
                id={3}
                isSelected={false}
                lastName="Manning"
                telephone="+63 672 620 4807"
/>,
              <ClientRow
                company="Brightbean"
                country="Sweden"
                firstName="Orlan"
                id={4}
                isSelected={false}
                lastName="Westbrook"
                telephone="+46 709 539 1100"
/>,
              <ClientRow
                company="Wikizz"
                country="Indonesia"
                firstName="Ivory"
                id={5}
                isSelected={false}
                lastName="Plaister"
                telephone="+62 443 276 2504"
/>,
              <ClientRow
                company="Avamm"
                country="China"
                firstName="Lindsey"
                id={6}
                isSelected={false}
                lastName="Revie"
                telephone="+86 341 167 4508"
/>,
              <ClientRow
                company="Mita"
                country="Indonesia"
                firstName="Constantin"
                id={7}
                isSelected={false}
                lastName="Tyer"
                telephone="+62 201 966 3247"
/>,
              <ClientRow
                company="Quimm"
                country="Poland"
                firstName="Letizia"
                id={8}
                isSelected={false}
                lastName="Emeney"
                telephone="+48 944 486 3972"
/>,
              <ClientRow
                company="Topicshots"
                country="Portugal"
                firstName="Rivkah"
                id={9}
                isSelected={false}
                lastName="Sabberton"
                telephone="+351 675 609 7738"
/>,
              <ClientRow
                company="Browsedrive"
                country="Dominican Republic"
                firstName="Jaine"
                id={10}
                isSelected={false}
                lastName="Hubback"
                telephone="+1 411 500 2183"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "company": "Dynazzy",
                "country": "Nigeria",
                "firstName": "Mariejeanne",
                "id": 1,
                "isSelected": false,
                "lastName": "Lush",
                "telephone": "+234 494 203 0408",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "company": "Bluejam",
                "country": "Zimbabwe",
                "firstName": "Jonas",
                "id": 2,
                "isSelected": false,
                "lastName": "Jeannaud",
                "telephone": "+263 706 597 3567",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "function",
              "props": Object {
                "company": "Divape",
                "country": "Philippines",
                "firstName": "Raynell",
                "id": 3,
                "isSelected": false,
                "lastName": "Manning",
                "telephone": "+63 672 620 4807",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "function",
              "props": Object {
                "company": "Brightbean",
                "country": "Sweden",
                "firstName": "Orlan",
                "id": 4,
                "isSelected": false,
                "lastName": "Westbrook",
                "telephone": "+46 709 539 1100",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "function",
              "props": Object {
                "company": "Wikizz",
                "country": "Indonesia",
                "firstName": "Ivory",
                "id": 5,
                "isSelected": false,
                "lastName": "Plaister",
                "telephone": "+62 443 276 2504",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "function",
              "props": Object {
                "company": "Avamm",
                "country": "China",
                "firstName": "Lindsey",
                "id": 6,
                "isSelected": false,
                "lastName": "Revie",
                "telephone": "+86 341 167 4508",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "function",
              "props": Object {
                "company": "Mita",
                "country": "Indonesia",
                "firstName": "Constantin",
                "id": 7,
                "isSelected": false,
                "lastName": "Tyer",
                "telephone": "+62 201 966 3247",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "function",
              "props": Object {
                "company": "Quimm",
                "country": "Poland",
                "firstName": "Letizia",
                "id": 8,
                "isSelected": false,
                "lastName": "Emeney",
                "telephone": "+48 944 486 3972",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "9",
              "nodeType": "function",
              "props": Object {
                "company": "Topicshots",
                "country": "Portugal",
                "firstName": "Rivkah",
                "id": 9,
                "isSelected": false,
                "lastName": "Sabberton",
                "telephone": "+351 675 609 7738",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "function",
              "props": Object {
                "company": "Browsedrive",
                "country": "Dominican Republic",
                "firstName": "Jaine",
                "id": 10,
                "isSelected": false,
                "lastName": "Hubback",
                "telephone": "+1 411 500 2183",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
